02-getting-account-data/index.js

```diff
@@ -0,0 +1,38 @@
+import { SDK } from 'NEWSDKNAME'
+
+// SDK exposes all you need like anchor-link does, users don't need to import from core directly, if they need something they do
+// import { Account } from 'NEWSDKNAME'
+
+const sdk = new SDK({ url: 'https://eos.greymass.com' }) // loaded with default plugins (something like if (browser) { plugins = [anchor-browser, atomichub] })
+
+async function main() {
+    const session = await sdk.login('myapp') // login also restores existing session unlike anchor-link, can have option for forcing re-login
+
+    // Loads account
+    const account = await session.accounts[0].get()
+    // Or you can load the account from the account name with either
+    // const account = await session.accounts.get('teamgreymass')
+    // or
+    // const account = await Account.get('eos', 'teamgreymass')
+
+    // Loads account balance
+    const balance = await account.getBalance('EOS')
+
+    // Load account resources
+    const { cpu, net, ram } = await account.getResources()
+
+    // Load account permissions
+    const permissions = await account.getPermissions()
+
+    // Load account actions
+    const actions = await account.getActions()
+    // Or with options
+    // const actions = await account.getActions({ scope: 'rewards.gm', limit: 100, offset: 0 })
+
+    // Load account transactions
+    const transactions = await account.getTransactions()
```

Comment:
This seems redundant of `getActions`. There isn't a specific `get_transactions` API endpoint either to call for this.

Reply:
Could it use the `get_actions` endpoint to get a list of transactions and return those? I guess it might not be that useful though, one could always do: `account.getActions[0].transaction` to get some transaction information.

Reply:
The `get_actions` API endpoint returns transactions already I thought. Are you just trying to retrieve an actual eosio-core Transaction object from that feed?

Reply:
I wonder if it is worth it to separate anything history related out to a separate module and not deal with it here. We are going to have to consider things like users wanting to use different history solutions. I'm leaning towards saying that the account object should not access anything that cannot be provided by the chain_api

Reply:
Yeah, exactly. The method would return eosio core Transaction objects.

Reply:
We removed this part of the code, so I'm resolving the conversation. We'll need some solution for history - but not here!